pkg/build/controller/buildpod/controller.go

```diff
@@ -121,6 +121,9 @@ func (bc *BuildPodController) HandlePod(pod *kapi.Pod) error {
 	nextStatus := build.Status.Phase
 	currentReason := build.Status.Reason
 
+	// if the build is marked failed, the build status reason has already been
+	// set (probably by the build pod itself), so don't do any updating here
+	// or we'll overwrite the correct value.
```

Comment:
Should we have utility functions that delineate which non-successful BuildPhases are set by the build pod (where it is *currently* just Failed) and which are set outside of the build pod (i.e. BuildPhaseError), in case those two lists change over time?